Docs for woocommerce additional token support

ndeet opened the pull request:
this is draft version for an upcoming new feature discussed here: https://github.com/btcpayserver/woocommerce-plugin/pull/59

pavlenex commented:
I think this is more suitable as FAQ then a separate page. This is not something that needs that prominent displaying imo, people who need this will probably search through FAQ-Integrations : WooCommerce where we have similar tutorials.

ndeet commented:
@pageman Ok got it, makes sense, I was not aware of the WooCommerce FAQ. The other WooCommerce FAQ entries are quite short and either link to WooCommerce main doc or external guide. Do you want it directly integrated on the FAQ page? Imo it is a bit too long for that? Or should I keep the separate page as is and link it in the WooCommerce FAQ but not in the menu?

Zaxounette commented:
First of all, congratz because I understand everything you explained in your doc, and I already like the possibilities that will be offered with this feature =)

As for a "review", just a few things:
* I concur with Pav, this doc could be placed under [WooCommerce-FAQ](https://docs.btcpayserver.org/FAQ/FAQ-Integrations/#woocommerce-faq-2) for the time being.

* I'd just like to see additional wording in your use-case examples to explicit the fact you are stating examples for this particular new feature.
Things in the likes of rewording the section title to `Use-case examples with Liquid Assets` and including `Liquid Asset(s)` in the use-cases paragraphes for example. If you come up with better wording than that, be my guest.

ndeet commented:
Thanks @Zaxounette 

> I concur with Pav, this doc could be placed under WooCommerce-FAQ for the time being.

Ok, should I put it on the existing page or leave this separate page and put a link in there (I think this makes the most sense otherwise it would go too long). What do you think?

> I'd just like to see additional wording in your use-case examples to explicit the fact you are stating examples for this particular new feature.
Things in the likes of rewording the section title to Use-case examples with Liquid Assets and including Liquid Asset(s) in the use-cases paragraphes for example. If you come up with better wording than that, be my guest.

About the naming of things I could use some help/suggestions here. It is not only Liquid Assets based, all Alts are supported. So you can add Monero, ETH, LTC and any other supported Altcoin as a separate payment method in WooCommerce. Afaik the ETH support also allows ERC20 tokens, so you could use them similar as Liquid Tokens but I'm not sure if those ERC20 tokens allow for zero decimals and therefore can be used as promotional tokens the same way as Liquid Assets.

Zaxounette commented on docs/WooCommerceAdditionalTokens.md:
```diff
@@ -0,0 +1,143 @@
+# WooCommerce Additional Tokens
+
+With using the additional token configuration you will be able to have separate payment methods for each configured token/currency. This means you can have separate payment methods for BTC, Lightning Network, LTC, ETH (and ERC20 tokens), Liquid assets, ... you name it. This allows you to issue and use [Liquid Assets](https://help.blockstream.com/hc/en-us/articles/900001543146-What-are-Liquid-asse ...
```
```suggestion
With using the additional token configuration you will be able to have separate payment methods for each configured Currency, Asset, Altcoin or Token. This means you can have separate payment methods for BTC, Lightning Network, LTC, ETH (and ERC20 tokens), Liquid assets, ... you name it. This allows you to issue and use [Liquid Assets](https://help.blockstream.com/hc/en-us/articles/900001543146-What-are-Liquid-assets-) as coupons or vouchers, see more details below.)
```

Zaxounette commented on docs/WooCommerceAdditionalTokens.md:
```diff
@@ -0,0 +1,143 @@
+// todo: add screenshot
+
+### Use-case 2: (Promotion) products should have free shipping
+This is doable with woocommerce out of the box (no plugins needed):
```
```suggestion
This adds the ability to offer free shipping when a customer pays with a selected Currency, Asset, Altcoin or Token.
This is doable with woocommerce out of the box (no plugins needed):
```

Zaxounette commented on docs/WooCommerceAdditionalTokens.md:
```diff
@@ -0,0 +1,143 @@
+   // todo: add screenshot
+
+### Use-case 3: limit product payment methods
+E.g. allow only a special token payment for promotional product(s)
```
```suggestion
E.g. allow only a certain Currency, Asset, Altcoin or Token to be used as payment for promotional product(s)
```

Zaxounette commented on docs/WooCommerceAdditionalTokens.md:
```diff
@@ -0,0 +1,143 @@
+// todo: add screenshot
+
+### Use-case 4: discount per payment method
+
```
```suggestion
Adds the ability to offer a discount when customer uses a selected Currency, Asset, Altcoin or Token as payment.

```

Zaxounette commented on docs/WooCommerceAdditionalTokens.md:
```diff
@@ -0,0 +1,143 @@
+
+### Use-case 5: make sure promotional products can only be purchased exclusively
+
+This is needed because promotion token payment methods need to override the product price with 1 (per quantity). Otherwise a user could mix normal products with promo products during checkout, which you want to avoid.
```
```suggestion
This is needed because payment methods based on a selected Currency, Asset, Altcoin or Token need to override the product price with 1 (per quantity). Otherwise a user could mix normal products with promo products during checkout, which you want to avoid.
```

ndeet replied:
I will copy this but enrich it a bit with more information to avoid confusion.

Zaxounette commented on docs/WooCommerceAdditionalTokens.md:
````diff
@@ -0,0 +1,143 @@
+```
+
+### Use-case 6: Limit the checkout of only 1 piece of a product
+
````
```suggestion
Adds the ability to limit the number of a Currency, Asset, Altcoin or Token that can be spent by a customer in a single checkout. 

Useful for coupon-style promotions that are limited to one discount per checkout.

```

Zaxounette commented on docs/WooCommerceAdditionalTokens.md:
```diff
@@ -0,0 +1,143 @@
+// todo: add screenshot
+
+
+## Common WooCommerce use-cases with other plugins
```
```suggestion
## Common WooCommerce use-cases using the Additionnal Token Support feature
```

Zaxounette reviewed (commented):
> Ok, should I put it on the existing page or leave this separate page and put a link in there (I think this makes the most sense otherwise it would go too long). What do you think?

Just put it on the existing page I linked.
If it's too long, we'll tinker with it afterwards. For the time being, the important issue is having this documentation in when the feature is available.
Don't forget to undo your changes in config.js =)

> About the naming of things I could use some help/suggestions here. It is not only Liquid Assets based, all Alts are supported. So you can add Monero, ETH, LTC and any other supported Altcoin as a separate payment method in WooCommerce. Afaik the ETH support also allows ERC20 tokens, so you could use them similar as Liquid Tokens but I'm not sure if those ERC20 tokens allow for zero decimals and therefore can be used as promotional tokens the same way as Liquid Assets.

Okay I get the gist.
ETH support will indeed enable ERC20 (and other ERC's BTW). ERC tokens have decimals, that might be tricky in production, but it doesn't concern this doc.

I proposed wording changes below for you.
Mostly just adding `Currency, Asset, Altcoin and Token` to sentences.
If one doesn't fit the intended context or if grammar is off, feel free to correct.

ndeet commented:
@Zaxounette awesome, thank you very much, your suggestion makes a lot of sense 👍

ndeet commented:
@Zaxounette fyi, moved the doc to existing file

Zaxounette reviewed (approved):
LGTM.
I'll just ping @britttttk to see if she has any other concerns.

ndeet commented:
still todo on my side (will do today or tomorrow)
- add screenshots

Zaxounette commented:
@ndeet Congratz on the launch and the Blockstream showcase of your features ! 

Do you have any other things to add to this doc ?

ndeet commented:
@Zaxounette thanks a lot 🎉 

Yes, there is a new checkbox that needs some documentation. Will add it in the next few days.

Zaxounette commented:
> Yes, there is a new checkbox that needs some documentation. Will add it in the next few days.

Noted !

Zaxounette commented:
Bump.

ndeet commented:
@Zaxounette thanks for the reminder 😇 

Added the missing documentation and rebased and fixed a conflict in package.json so the PR is up to date with current master.

Zaxounette commented on docs/FAQ/FAQ-Integrations.md:
```diff
@@ -84,3 +85,154 @@ function get_order_id_from_custom_order_style($orderid){
+
+#### Requirements
+- all tokens you configure on the WooCommerce side need to be available in your store on BTCPay Server side
+- to use promotional tokens you need to have [Liquid Assets plugin](https://https://github.com/btcpayserver/btcpayserver-plugins) installed on BTCPay Server
```
The `Liquid Assets plugin` link doesn't work.
Might be a private repo ? Would be surprised, but maybe.

Zaxounette replied:
Never mind, i'm a dumbass.
Here's the fix:

```suggestion
- to use promotional tokens you need to have [Liquid Assets plugin](https://github.com/btcpayserver/btcpayserver-plugins) installed on BTCPay Server
```

ndeet replied:
Nice catch 👍

Zaxounette commented on docs/FAQ/FAQ-Integrations.md:
```diff
@@ -84,3 +85,154 @@ function get_order_id_from_custom_order_style($orderid){
+![Each additional additional token available as payment token](../img/woocommerce/woocommerce_at_payment-methods.png)
+
+###### Setting: Additional tokens: Enforce payment tokens
+The default payment method of BTCPay Server (Bitcoin) will **not** enforce any configured Currency, Asset, Altcoin or Token. This means when you have the default payment method "Bitcoin" enabled the user can select all the configured Currency, Asset, Altcoin or Tokens (which have an exchange rate) on the BTCPay Server payment page. You may not want this but enforce/limit which payment options wil ...
```
```suggestion
The default payment method of BTCPay Server (Bitcoin) will **not** enforce any configured Currency, Asset, Altcoin or Token. This means when you have the default payment method "Bitcoin" enabled the user can select all the configured Currency, Asset, Altcoin or Tokens (which have an exchange rate) on the BTCPay Server payment page. You may not want this but enforce/limit which payment options will be available. By selecting this checkbox only the Currency, Asset, Altcoin or Tokens of type "payment" listed in the setting [Setting: Additional token configuration](#setting-additional-token-configuration)
```

Typo.

ndeet replied:
👍

Zaxounette commented on docs/FAQ/FAQ-Integrations.md:
```diff
@@ -84,3 +85,154 @@ function get_order_id_from_custom_order_style($orderid){
 </details>
+
+### How to configure Additional Token Support
+
```
```suggestion

::: tip Note
The Wordpress and Woocommerce plugins used in this integrations are neither endorsed nor have they been thoroughly verified or vetted by the BTCPay Server team. Use them at your own risks.

```

Disclaimer. You never know.

ndeet replied:
Good idea, thanks!

Zaxounette reviewed (changes requested):
Looks good ! 
A few things though below.

- A disclaimer
- A typo
- A broken link

Zaxounette commented:
I'll also ping @dennisreimann to confirm that the package.json change is good for him.

ndeet reviewed (commented):
thanks @Zaxounette 🤗

dennisreimann commented:
ACK, good work!

Zaxounette commented:
@dennisreimann @britttttk PUNCH IT

Kukks commented:
...Why was this merged? It is for an unmerged feature in the woocommerce plugin.

dennisreimann commented:
@Kukks oh sorry, this was my fault. I took Zax comment as a request to merge it. Will try to revert.

Zaxounette commented:
It was a request to merge, you did understand me correctly Dennis.

This in on me.

Kukks commented:
> @Kukks oh sorry, this was my fault. I took Zax comment as a request to merge it. Will try to revert.

Nah don't worry. You're  now tasked with reviewing the PR in woocommerce plugin with me @dennisreimann  though :D

dennisreimann commented:
Expediting things here 😬😆

dennisreimann commented:
Now that I've set it up using the guide from the FAQ I think it'd be helpful to add a link/reference of where to find/lookup the token symbol in BTCPay Server.

I also think that structure-wise this would work better as a subpage for or included on the WooCommerce integration page. It's rather long and imho it'd work better there than in the FAQs.

Zaxounette commented:
> find/lookup the token symbol in BTCPay Server

Or simply say to use the 3 (or 4) letter trading ticker ? Since that's what is used ? Coz I don't know of a "list" of tokens inside the UI.

Subpage means subnav. We already have a ton.
If we create a subpage for one plugin, we'll create a precedent and have to make others for other plugins.
Not really looking forward to that.

As for FAQ or woocommerce plain and simple, i'll come up with something.

dennisreimann commented:
> Or simply say to use the 3 (or 4) letter trading ticker ? Since that's what is used ? Coz I don't know of a "list" of tokens inside the UI.

I think there's a bit more to it, e.g. when configuring the Lightning payment method. I wasn't sure where the `BTC_OFFCHAIN` part was coming from.
